app/Actions/Photo/Create.php

```diff
@@ -198,17 +201,17 @@ protected function initParentId($albumID = null)
 		/** @var AlbumFactory */
 		$factory = resolve(AlbumFactory::class);
 		if (!empty($albumID)) {
-			$album = $factory->make($albumID);
-
-			if ($album->is_tag_album()) {
-				throw new JsonError('Sorry, cannot upload to Tag Album.');
-			}
-
-			if (!$album->is_smart()) {
-				$this->strategyParameters->album = $album; // we save it so we don't have to query it again later
+			$album = $factory->findOrFail($albumID);
+
+			if ($album instanceof Album) {
+				// we save it so we don't have to query it again later
+				$this->strategyParameters->album = $album;
+			} elseif ($album instanceof PublicAlbum) {
+				$this->strategyParameters->is_public = true;
+			} elseif ($album instanceof StarredAlbum) {
+				$this->strategyParameters->is_starred = true;
 			} else {
-				$this->strategyParameters->public = ($album->id == 'public');
-				$this->strategyParameters->star = ($album->id == 'starred');
+				throw new JsonError('This album does not support uploading');
```

Comment:
This could be localized with the Lang facade, but I am fine with not doing it. :)

Reply:
I already have another PR in the pipeline which deals with return types and correct HTTP status codes. This future PR will be much easier than the last two PRs.

Anyway, this is only an intermediate fix. Stay tuned for the upcoming PR. ;-)